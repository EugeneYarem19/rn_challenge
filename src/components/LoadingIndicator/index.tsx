import React from "react";
import { ActivityIndicator, ActivityIndicatorProps, } from "react-native";

import { palette, } from "@theme";

export const LoadingIndicator: React.FC<ActivityIndicatorProps> = ({ color = palette.accentColor, size = "large", }) => (
  <ActivityIndicator size={size} color={color} />
);
