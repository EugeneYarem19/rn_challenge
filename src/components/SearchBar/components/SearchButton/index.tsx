import React from "react";
import { View, } from "react-native";
import { Button, } from "react-native-material-ui";

import { styles, } from "./styles";

interface SearchCallback {
  (): void;
}

interface Props {
  onPress: SearchCallback;
}

export const SearchButton: React.FC<Props> = ({ onPress, }): JSX.Element => (
  <View testID="SearchButton">
    <Button icon="search" text="" onPress={onPress} style={{ container: styles.container, }} />
  </View>
);
