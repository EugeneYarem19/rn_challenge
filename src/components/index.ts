export * from "./LoadingIndicator";
export * from "./SearchBar";
