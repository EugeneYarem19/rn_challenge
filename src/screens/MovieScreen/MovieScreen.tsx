import React from "react";
import { View, } from "react-native";

import { Movie, } from "@redux";

import { styles, } from "./styles";

interface Props extends Movie {
    defaultTitle: string;
}

export const MovieScreen: React.FC<Props> = (): JSX.Element => <View style={styles.screen} />;
