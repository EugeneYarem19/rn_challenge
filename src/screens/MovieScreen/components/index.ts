export * from "./DetailsItem";
export * from "./MovieScreenComponent";
export * from "./Rating";
