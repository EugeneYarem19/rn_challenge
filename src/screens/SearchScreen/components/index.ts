export * from "./SearchResultItem";
export * from "./SearchScreen";
