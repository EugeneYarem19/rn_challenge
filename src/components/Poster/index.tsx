import React from "react";
import ProgressImage from "react-native-image-progress";
import { Icon, } from "react-native-material-ui";
import { View, } from "react-native";

import { LoadingIndicator, } from "../LoadingIndicator";

import { IPoster, ImageSize, } from "./types";
import { styles, } from "./styles";
import { usePoster, } from "./hooks";

export const Poster: React.FC<IPoster> = ({
  poster,
  size = ImageSize.small,
  smallOnEmptyPoster = true,
  smallOnError = true,
  style,
}) => {
  const sizeStyle = size === ImageSize.small ? styles.small : styles.large;
  const { loadError, onError, } = usePoster();

  const renderPlaceholder = (additionalSizedStyle: typeof style) => (
    <View style={[styles.placeholder, additionalSizedStyle, style,]}>
      <Icon name="wallpaper" size={50} />
    </View>
  );

  const renderPosterIcon = () => {
    if (loadError) {
      return renderPlaceholder(smallOnError ? styles.small : sizeStyle);
    } else if (!poster) {
      return renderPlaceholder(smallOnEmptyPoster ? styles.small : sizeStyle);
    } else {
      return (
        <ProgressImage
          source={{ uri: poster, }}
          indicator={LoadingIndicator}
          onError={onError}
          resizeMode="contain"
          style={[styles.poster, sizeStyle, style,]}
          threshold={0}
        />
      );
    }
  };

  return renderPosterIcon();
};
