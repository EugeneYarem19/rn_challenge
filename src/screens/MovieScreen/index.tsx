import React from "react";
import { RouteProp, } from "@react-navigation/native";
import { useSelector, } from "react-redux";

import { Movie, State, } from "@redux";
import { RootStackParamList, } from "@navigation";

import { MovieScreen as MovieScreenComponent, } from "./MovieScreen";

type ScreenRouteProp = RouteProp<RootStackParamList, "Movie">;

type Props = {
  route: ScreenRouteProp;
};

export const MovieScreen: React.FC<Props> = ({ route, }): JSX.Element => {
  const foundMovies = useSelector((state: State) => state.foundMovies);

  const { id, title, } = route.params;

  const movie = foundMovies.find((item) => item.id === id) as Movie;

  return <MovieScreenComponent {...movie} defaultTitle={title} />;
};
