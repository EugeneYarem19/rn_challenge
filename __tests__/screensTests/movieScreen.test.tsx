import React from "react";
import configureStore from "redux-mock-store";
import { Provider, } from "react-redux";

import { MovieScreen, } from "@screens";
import { DetailsItem, Rating, } from "@src/screens/MovieScreen/components";

import { mockedData, } from "@tests/__mocks__";
import { snapshotTest, } from "@tests/__utils__";

const mockStore = configureStore([]);
const props: any = {};

describe("MovieScreen tests", () => {
  test("DetailsItem must renders correctly", () => {
    snapshotTest(<DetailsItem title="title" value="value" />);
  });

  test("Rating must renders correctly", () => {
    snapshotTest(<Rating ratings={[{ source: "source", value: "value", },]} />);
  });

  test("MovieScreen must renders correctly with all props", () => {
    const store = mockStore({
      isFetchingMovie: false,
      foundMovies: [
        {
          id: mockedData.id,
          poster: "http://",
          title: "Title",
          genre: "Genre",
          director: "Director",
          fullPlot: "Plot",
          cast: "Cast",
          ratings: [{ source: "source", value: "value", },],
        },
      ],
    });
    snapshotTest(
      <Provider store={store}>
        <MovieScreen
          {...props}
          route={{ params: { id: mockedData.id, title: mockedData.title, }, }}
        />
      </Provider>
    );
  });

  test("MovieScreen must display loading indicator", () => {
    const store = mockStore({
      isFetchingMovie: true,
      foundMovies: [
        {
          id: mockedData.id,
          poster: "http://",
          title: "Title",
        },
      ],
    });
    snapshotTest(
      <Provider store={store}>
        <MovieScreen
          {...props}
          route={{ params: { id: mockedData.id, title: mockedData.title, }, }}
        />
      </Provider>
    );
  });
});
