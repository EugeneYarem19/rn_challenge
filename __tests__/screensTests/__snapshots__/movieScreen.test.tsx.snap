// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieScreen tests DetailsItem must renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ccc",
        "fontSize": 21,
        "fontWeight": "bold",
        "marginRight": 5,
      }
    }
  >
    title: 
  </Text>
  <Text
    style={
      Object {
        "color": "#ccc",
        "flex": 1,
        "fontSize": 20,
      }
    }
  >
    value
  </Text>
</View>
`;

exports[`MovieScreen tests MovieScreen must display loading indicator 1`] = `
<RCTScrollView
  bounces={false}
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "#131d3d",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#ccc",
          "fontFamily": "Arial",
          "fontSize": 32,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Title
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#131d3d",
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 20,
            "height": 1027.6,
            "maxHeight": 443,
            "maxWidth": 300,
            "width": 734,
          },
          Object {
            "backgroundColor": "#1e3166",
            "borderColor": "#1e3166",
            "borderWidth": 4,
            "marginBottom": 15,
            "marginTop": 15,
          },
        ]
      }
    >
      <Image
        onError={[Function]}
        onLoad={[Function]}
        onLoadEnd={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "http://",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#e59f2b"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
    <ActivityIndicator
      animating={true}
      color="#e59f2b"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</RCTScrollView>
`;

exports[`MovieScreen tests MovieScreen must renders correctly with all props 1`] = `
<RCTScrollView
  bounces={false}
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "#131d3d",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#ccc",
          "fontFamily": "Arial",
          "fontSize": 32,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Title
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#131d3d",
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 20,
            "height": 1027.6,
            "maxHeight": 443,
            "maxWidth": 300,
            "width": 734,
          },
          Object {
            "backgroundColor": "#1e3166",
            "borderColor": "#1e3166",
            "borderWidth": 4,
            "marginBottom": 15,
            "marginTop": 15,
          },
        ]
      }
    >
      <Image
        onError={[Function]}
        onLoad={[Function]}
        onLoadEnd={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "http://",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#e59f2b"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 21,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      >
        Ratings: 
      </Text>
      <View>
        <View>
          <Text
            style={
              Object {
                "color": "#ccc",
                "fontSize": 20,
                "marginRight": 5,
                "textDecorationLine": "underline",
              }
            }
          >
            source
          </Text>
          <Text
            style={
              Object {
                "color": "#ccc",
                "flex": 1,
                "fontSize": 21,
                "fontWeight": "bold",
                "marginVertical": 3,
              }
            }
          >
            value
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 21,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      >
        Genre: 
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "flex": 1,
            "fontSize": 20,
          }
        }
      >
        Genre
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 21,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      >
        Director: 
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "flex": 1,
            "fontSize": 20,
          }
        }
      >
        Director
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 21,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      >
        Cast: 
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "flex": 1,
            "fontSize": 20,
          }
        }
      >
        Cast
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 21,
            "fontWeight": "bold",
            "marginRight": 5,
          }
        }
      >
        Plot: 
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "flex": 1,
            "fontSize": 20,
          }
        }
      >
        Plot
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`MovieScreen tests Rating must renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ccc",
        "fontSize": 21,
        "fontWeight": "bold",
        "marginRight": 5,
      }
    }
  >
    Ratings: 
  </Text>
  <View>
    <View>
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 20,
            "marginRight": 5,
            "textDecorationLine": "underline",
          }
        }
      >
        source
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "flex": 1,
            "fontSize": 21,
            "fontWeight": "bold",
            "marginVertical": 3,
          }
        }
      >
        value
      </Text>
    </View>
  </View>
</View>
`;
